Integration/AnuraIntegration.php

```diff
@@ -0,0 +1,154 @@
+
+    /**
+     * @return string
+     */
+    public function getName()
+    {
+        return 'Anura';
+    }
+
+    /**
+     * @return string
+     */
+    public function getDisplayName()
+    {
+        return 'Anura Suspicious Check';
+    }
+
+    /**
+     * @return string
+     */
+    public function getAuthenticationType()
+    {
+        return 'keys';
+    }
+
+    /**
+     * @return AnuraModel|AbstractCommonModel
+     */
+    public function getModel()
+    {
+        if (!isset($this->integrationModel)) {
+            $this->integrationModel = $this->factory->getModel('enhancer.anura');
+            $this->integrationModel->setup($this);
+        }
+
+        return $this->integrationModel;
+    }
+
+    /**
+     * @return array
+     */
+    public function getRequiredKeyFields()
+    {
+        return [
+            'endpoint' => $this->translator->trans('mautic.enhancer.integration.anura.endpoint.label'),
+            'instance' => $this->translator->trans('mautic.enhancer.integration.anura.instance.label'),
+        ];
+    }
+
+    /**
+     * @return array
+     */
+    public function getSupportedFeatures()
+    {
+        return ['push_lead'];
+    }
+
+    /**
+     * @param \Symfony\Component\Form\FormBuilderInterface $builder
+     * @param array                                        $data
+     * @param string                                       $formArea
+     */
+    public function appendToForm(&$builder, $data, $formArea)
+    {
+        if ('features' === $formArea) {
+            $builder->add(
+                'default_user_agent',
+                TextType::class,
+                [
+                    'label'      => $this->translator->trans('mautic.enhancer.integration.anura.user_agent.label'),
+                    'data'       => isset($data['default_user_agent']) ? $data['default_user_agent'] : '',
+                    'required'   => true,
+                    'label_attr' => ['class' => 'control-label'],
+                    'attr'       => [
+                        'class'   => 'form-control',
+                        'tooltip' => $this->translator->trans('mautic.enhancer.integration.anura.user_agent.tooltip'),
+                    ],
+                ]
+            );
+
+            $this->appendNonFreeFields($builder, $data, $formArea, true);
+        }
+    }
+
+    /**
+     * @return array|mixed
+     */
+    protected function getEnhancerFieldArray()
+    {
+        return [
+            'anura_result' => [
+                'label' => $this->translator->trans('mautic.enhancer.integration.anura.anura_result.label'),
+                'group' => 'enhancement',
+            ],
+        ];
+    }
+
+    /**
+     * @param Lead $lead
+     *
+     * @return bool
+     */
+    public function doEnhancement(Lead &$lead)
+    {
+        if (!empty($lead)) {
+            $ipAddress = $this->request->server->get('REMOTE_ADDR');
+            $userAgent = $this->request->server->get('HTTP_USER_AGENT');
```

Comment:
We should be checking the user_agent (from UTM pivot data) and IP address from the lead, not of the incoming request, which usually has no bearing on the lead that came from an API/Source..